config/initializers/zeitwerk.rb

```diff
@@ -9,3 +9,27 @@
 Rails.autoloaders.main.collapse(Rails.root.join("lib/manageiq/reporting/charting"))
 Rails.autoloaders.main.collapse(Rails.root.join("lib/ansible/runner/credential"))
 Rails.autoloaders.main.collapse(Rails.root.join("lib/pdf_generator"))
+
+excluded_from_eager_load = %w[
+  ApplicationRecord
+  MiqDecorator
+  OrchestrationStack::Status
+  ResourcePool
+]
+
+DescendantLoader.instance.discovered_parent_child_classes.each do |parent, children|
+  next if excluded_from_eager_load.include?(parent)
+
+  puts "Registering on_load for class: #{parent}"
```

Comment:
I know this is for debug but I wonder if we should keep it and use $log.debug instead. (Same comment for the other puts lines)